sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceivedMessage.java

```diff
@@ -50,50 +60,56 @@
     }
 
     /**
-     * The representation of message as defined by AMQP protocol.
-     *
-     * @see <a href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#section-message-format">
-     *     Amqp Message Format.</a>
+     * Gets the set of free-form {@link ServiceBusReceivedMessage} properties which may be used for passing metadata
+     * associated with the {@link ServiceBusReceivedMessage} during Service Bus operations. A common use-case for
+     * {@code properties()} is to associate serialization hints for the {@link #getBody()} as an aid to consumers
+     * who wish to deserialize the binary data.
      *
-     * @return the {@link AmqpAnnotatedMessage} representing amqp message.
+     * @return Application properties associated with this {@link ServiceBusReceivedMessage}.
+     * @see ServiceBusMessage#getApplicationProperties()
      */
-    public AmqpAnnotatedMessage getRawAmqpMessage() {
-        return amqpAnnotatedMessage;
+    public Map<String, Object> getApplicationProperties() {
+        return amqpAnnotatedMessage.getApplicationProperties();
     }
 
     /**
-     * Gets the actual payload/data wrapped by the {@link ServiceBusReceivedMessage}.
+     * Gets the payload wrapped by the {@link ServiceBusReceivedMessage}.
      *
      * <p>The {@link BinaryData} wraps byte array and is an abstraction over many different ways it can be represented.
-     * It provides many convenience API including APIs to serialize/deserialize object.
+     * It provides convenience APIs to serialize/deserialize the object.</p>
+     *
      * <p>
      * If the means for deserializing the raw data is not apparent to consumers, a common technique is to make use of
      * {@link #getApplicationProperties()} when creating the event, to associate serialization hints as an aid to
      * consumers who wish to deserialize the binary data.
      * </p>
      *
-     * @return A byte array representing the data.
+     * @return Binary data representing the payload.
+     * @see ServiceBusMessage#getBody()
      */
     public BinaryData getBody() {
         final AmqpMessageBodyType bodyType = amqpAnnotatedMessage.getBody().getBodyType();
         switch (bodyType) {
             case DATA:
-                return BinaryData.fromBytes(amqpAnnotatedMessage.getBody()
-                    .getData().stream().findFirst().get());
+                final Optional<byte[]> payload = amqpAnnotatedMessage.getBody().getData().stream().findFirst();
```

Comment:
I missed it to change to `return BinaryData.fromBytes(amqpAnnotatedMessage.getBody().getFirstData())`

BinaryData will always have data, a null is converted into empty byte array inside BinaryData, So we do not have to 
 check for `payload.isPresent()`.


BinaryData  store data in byte[] variable. so calling `getFirstData` will be efficient.

Reply:
I had an IntelliJ warning about this. I think it's defensive programming. You only know this because you implemented it. We could change it.

Reply:
We already have call to `getFirstData` in `ServiceBusMessage` and I did not see any intellij warning. 
This API `getFirstData` was added for one of the reason 
1. No need to create unnecessarily create `List`  when we will have one byte[] always.

In SBRM.getBody() have return type BinaryData, so it is one byte [] always. So there is no harm in using `getFirstData()`. And you do not need to depend/assume anything how it is implemented internally. If you want to check for null when `getFirstData()` returns, that is fine. 

Think about received message performance , when every message create aList of one byte[] for 95% of users.